diff --git a/erts/configure.ac b/erts/configure.ac
index ab2ee78a..4c3a9b89 100644
--- a/erts/configure.ac
+++ b/erts/configure.ac
@@ -23,7 +23,7 @@ dnl autoconf convention that should be cleaned up.
 
 AC_INIT
 AC_CONFIG_SRCDIR([emulator/beam/erl_process.c])
-AC_PREREQ([2.71])
+AC_PREREQ([2.69])
 
 m4_include([otp.m4])
 
@@ -1628,7 +1628,6 @@ saved_cppflags=$CPPFLAGS
 CPPFLAGS="$CPPFLAGS $EMU_THR_DEFS"
 
 AC_HEADER_DIRENT
-AC_CHECK_INCLUDES_DEFAULT
 
 AC_HEADER_SYS_WAIT
 AC_CHECK_HEADERS_ONCE([sys/time.h])
diff --git a/lib/common_test/configure.ac b/lib/common_test/configure.ac
index 490a6432..46077886 100644
--- a/lib/common_test/configure.ac
+++ b/lib/common_test/configure.ac
@@ -1,5 +1,5 @@
 AC_INIT
-AC_PREREQ([2.71])
+AC_PREREQ([2.69])
 
 m4_include([otp.m4])
 
diff --git a/lib/common_test/test_server/configure.ac b/lib/common_test/test_server/configure.ac
index da5c1b16..c900dc99 100644
--- a/lib/common_test/test_server/configure.ac
+++ b/lib/common_test/test_server/configure.ac
@@ -21,7 +21,7 @@ dnl
 
 AC_INIT
 AC_CONFIG_SRCDIR([conf_vars.in])
-AC_PREREQ([2.71])
+AC_PREREQ([2.69])
 
 AC_CANONICAL_HOST
 
diff --git a/lib/crypto/configure.ac b/lib/crypto/configure.ac
index 3f595ad4..36dfdf5e 100644
--- a/lib/crypto/configure.ac
+++ b/lib/crypto/configure.ac
@@ -24,7 +24,7 @@ dnl define([AC_CACHE_SAVE], )dnl
 
 AC_INIT
 AC_CONFIG_SRCDIR([vsn.mk])
-AC_PREREQ([2.71])
+AC_PREREQ([2.69])
 
 m4_include([otp.m4])
 
@@ -806,7 +806,7 @@ AS_CASE(["$with_ssl_rpath"],
             # Use ssl runtime library paths set by --with-ssl-rpath
             AC_MSG_CHECKING([whether runtime library path can be set])
 
-            AS_IF([test $enable_dynamic_ssl = no],
+            AS_IF([test "$enable_dynamic_ssl" = no],
                   [
                       AC_MSG_ERROR([runtime library path requested by user, but user also requested static linking])
                   ],
diff --git a/lib/erl_interface/configure.ac b/lib/erl_interface/configure.ac
index d24e3f48..9854c2b4 100644
--- a/lib/erl_interface/configure.ac
+++ b/lib/erl_interface/configure.ac
@@ -27,7 +27,7 @@ dnl m4_define(EI_VERSION,regexp(m4_include(VERSION),[version \([-.0-9A-Za-z]+\)]
 
 AC_INIT
 AC_CONFIG_SRCDIR([include/ei.h])
-AC_PREREQ([2.71])
+AC_PREREQ([2.69])
 
 m4_include([otp.m4])
 
@@ -171,7 +171,6 @@ AC_SEARCH_LIBS([gethostbyname], [nsl])
 AC_SEARCH_LIBS([getpeername], [socket])
 
 # Checks for header files.
-AC_CHECK_INCLUDES_DEFAULT
 
 AC_HEADER_SYS_WAIT
 AC_CHECK_HEADERS([arpa/inet.h fcntl.h limits.h malloc.h netdb.h netinet/in.h stddef.h stdlib.h string.h sys/param.h sys/socket.h sys/select.h sys/time.h unistd.h sys/types.h sys/uio.h])
diff --git a/lib/megaco/configure.ac b/lib/megaco/configure.ac
index ce65e54a..b5f57efc 100644
--- a/lib/megaco/configure.ac
+++ b/lib/megaco/configure.ac
@@ -25,7 +25,7 @@ dnl define([AC_CACHE_SAVE], )dnl
 
 AC_INIT
 AC_CONFIG_SRCDIR([vsn.mk])
-AC_PREREQ([2.71])
+AC_PREREQ([2.69])
 
 m4_include([otp.m4])
 
diff --git a/lib/odbc/configure.ac b/lib/odbc/configure.ac
index dfc3a6ec..03f3fa74 100644
--- a/lib/odbc/configure.ac
+++ b/lib/odbc/configure.ac
@@ -25,7 +25,7 @@ dnl define([AC_CACHE_SAVE], )dnl
 dnl Process this file with autoconf to produce a configure script.
 AC_INIT
 AC_CONFIG_SRCDIR([c_src/odbcserver.c])
-AC_PREREQ([2.71])
+AC_PREREQ([2.69])
 
 m4_include([otp.m4])
 
@@ -112,7 +112,6 @@ AS_CASE(["$host_os"],
         ])
 
 dnl Checks for header files.
-AC_CHECK_INCLUDES_DEFAULT
 AC_PROG_EGREP
 
 AC_CHECK_HEADERS([fcntl.h netdb.h stdlib.h string.h sys/socket.h winsock2.h])
diff --git a/lib/snmp/configure.ac b/lib/snmp/configure.ac
index 8f4c1fd8..0077732a 100644
--- a/lib/snmp/configure.ac
+++ b/lib/snmp/configure.ac
@@ -25,7 +25,7 @@ define([AC_CACHE_SAVE], )dnl
 
 AC_INIT
 AC_CONFIG_SRCDIR([src/app/snmp.app.src])
-AC_PREREQ([2.71])
+AC_PREREQ([2.69])
 
 m4_include([otp.m4])
 
diff --git a/lib/wx/configure.ac b/lib/wx/configure.ac
index d49399c5..0909cee2 100644
--- a/lib/wx/configure.ac
+++ b/lib/wx/configure.ac
@@ -24,7 +24,7 @@ m4_include([otp.m4])
 
 AC_CONFIG_AUX_DIR([${ERL_TOP}/make/autoconf])
 
-AC_PREREQ([2.71])
+AC_PREREQ([2.69])
 
 ## Delete previous failed configure results
 if test -f ./CONF_INFO; then
diff --git a/make/configure.ac b/make/configure.ac
index e0d4103b..a4130a42 100644
--- a/make/configure.ac
+++ b/make/configure.ac
@@ -18,7 +18,7 @@ dnl limitations under the License.
 dnl
 dnl %CopyrightEnd%
 
-AC_PREREQ([2.71])dnl
+AC_PREREQ([2.69])dnl
 
 AC_INIT
 
diff --git a/otp_build b/otp_build
index 7b35b39f..7fed8449 100755
--- a/otp_build
+++ b/otp_build
@@ -19,7 +19,7 @@
 # %CopyrightEnd%
 #
 
-USE_AUTOCONF_VERSION=2.71
+USE_AUTOCONF_VERSION=2.69
 
 autoconf_aux_dirs="./erts/autoconf ./lib/common_test/test_server"
 
