From: Clint Byrum <clint@ubuntu.com>
Subject: resolve buildd failure "Function `erl_malloc' implicitly converted to pointer at legacy/erl_timeout.c:77"
Bug-Ubuntu: https://launchpad.net/bugs/778484

Index: erlang-14.b.4-dfsg-1ubuntu1/lib/erl_interface/src/legacy/erl_timeout.c
===================================================================
--- erlang-14.b.4-dfsg-1ubuntu1.orig/lib/erl_interface/src/legacy/erl_timeout.c	2011-12-15 16:39:50.958344893 -0800
+++ erlang-14.b.4-dfsg-1ubuntu1/lib/erl_interface/src/legacy/erl_timeout.c	2011-12-15 16:40:45.183171477 -0800
@@ -45,6 +45,7 @@
 
 #include "erl_interface.h"
 #include "erl_timeout.h"
+#include "erl_interface.h"
 
 typedef struct jmp_s {
   jmp_buf jmpbuf;
@@ -75,7 +76,7 @@
   t.it_value.tv_usec = (ms % 1000) * 1000;
 
   /* get a jump buffer and save it */
-  j = erl_malloc(sizeof(*j));
+  j = (jmp_t)erl_malloc(sizeof(*j));
   j->siginfo = s;
   push(j);
 
