diff --git a/erts/emulator/sys/unix/sys_time.c b/erts/emulator/sys/unix/sys_time.c
index eb2c47bb..3810ac26 100644
--- a/erts/emulator/sys/unix/sys_time.c
+++ b/erts/emulator/sys/unix/sys_time.c
@@ -263,11 +263,23 @@ sys_init_time(ErtsSysInitTimeResult *init_resp)
 
 #if defined(__linux__) && defined(OS_MONOTONIC_TIME_USING_CLOCK_GETTIME)
     if (vsn >= ERTS_MK_VSN_INT(2, 6, 33)) {
+#if defined(HAVE_CLOCK_GETTIME_MONOTONIC_RAW)
+       init_resp->have_corrected_os_monotonic_time = 0;
+       erts_sys_time_data__.r.o.os_monotonic_time =
+           clock_gettime_monotonic_raw;
+#if defined(OS_SYSTEM_TIME_USING_CLOCK_GETTIME)
+       erts_sys_time_data__.r.o.os_times =
+           clock_gettime_times_raw;
+#endif
+       init_resp->os_monotonic_time_info.clock_id =
+           "CLOCK_MONOTONIC_RAW";
+#else
 	erts_sys_time_data__.r.o.os_monotonic_time =
 	    clock_gettime_monotonic;
 #if defined(OS_SYSTEM_TIME_USING_CLOCK_GETTIME)
 	erts_sys_time_data__.r.o.os_times =
 	    clock_gettime_times;
+#endif
 #endif
     }
     else {
