diff --git a/lib/crypto/c_src/rand.c b/lib/crypto/c_src/rand.c
index 82bf2c74..ad09d60c 100644
--- a/lib/crypto/c_src/rand.c
+++ b/lib/crypto/c_src/rand.c
@@ -18,6 +18,8 @@
  * %CopyrightEnd%
  */
 
+#include <stdio.h>
+
 #include "rand.h"
 #include "bn.h"
 
@@ -36,8 +38,10 @@ ERL_NIF_TERM strong_rand_bytes_nif(ErlNifEnv* env, int argc, const ERL_NIF_TERM
 
     if ((data = enif_make_new_binary(env, bytes, &ret)) == NULL)
         goto err;
-    if (RAND_bytes(data, (int)bytes) != 1)
+    if (RAND_bytes(data, (int)bytes) != 1) {
+        ERR_print_errors_fp(stderr);
         goto err;
+    }
 
     ERL_VALGRIND_MAKE_MEM_DEFINED(data, bytes);
     return ret;
